### DISCUSSION TITLE
In vue3 what is a lib to work with charts?

### DISCUSSION THREAD

**Discussion by ewertonrfd** [quasarframework/quasar#10614] [General - Components / Directives / etc] [answered]
_Initially, if it's not the right place, I'm sorry._

But I haven't found discussions on the subject yet.
**In vue3 what is a lib to work with charts?**

In the libs I've seen (at least so far 😅 ) I haven't found installation for vue 3

**Comment by anazzani** [comment]
I use [vue3-apexcharts](https://github.com/apexcharts/vue3-apexcharts).

HTH.

**Comment by ewertonrfd** [comment]
Thanks @anazzani 

I'll create a project with it separate to test 😃  

**Ask a question, this part of the code:** 

```js
import ApexCharts from 'apexcharts';

app.config.globalProperties.$apexcharts = ApexCharts;

// Add this when into a TypeScript codebase
declare module '@vue/runtime-core' {
  interface ComponentCustomProperties {
    $apexcharts: typeof ApexCharts;
  }
}
```

**Is it optional or also necessary for use?**

**Comment by anazzani** [comment, marked as answer]
Not sure about Typescript, but this seems to be enough in my .vue files:
```
import VueApexCharts from 'vue3-apexcharts'
.
.
.
  components: {
    apexchart: VueApexCharts
  },
.
.
.
```
and then I can use the `<apexchart></apexchart>` tag in the html section.

**Reply by ewertonrfd** [reply]
Installation was smooth!

**Comment by ontwikkelfabriek** [comment]
@anazzani 

Don't you have to use a **boot** file to register `VueApexCharts` ?

**Reply by anazzani** [reply]
I don't think it's necessary.

Perhaps a boot file is more practical if you don't want to import the lib in each .vue file, but if you have only one or two pages showing charts you can skip the registration.

At least this is how I understand it: I'm not really an expert. :)

**Comment by ewertonrfd** [comment]
Hi @anazzani 

I am now using apexcharts 

But I am having a very strange problem.
To update chart data coming from an API I can't :(

The array of examples I saw and the one I generated are the same, I even compared them online (https://jsoneditoronline.org/)

```
[
  {
    "name": "South",
    "data": [
      [
        1486778400000,
        44
      ],
      [
        1486864800000,
        40
      ],
      [
        1486951200000,
        57
      ],
    ]
  }
]
```

```
[
  {
    "name": "Novos",
    "data": [
      [
        "set",
        2
      ],
      [
        "ago",
        2
      ],
      [
        "jul",
        2
      ],
    ]
  }
]
```

**Arrays are bigger (only more records)**

I only put a part, to show the base
**But the example array works**
seriesUsers.value = newSeries;

**but mine doesn't**
seriesUsers.value = a;

_The graph even changes_
But it doesn't show bars or anything (like it's invisible or something)

I did not understand 👎 

Thanks for the help, since now

**Comment by ewertonrfd** [comment]
After many attempts and head banging with the documentation, it worked.

Thank you all 
@anazzani 
@ontwikkelfabriek 

**Result of dashboard:**

![2021-09-17](https://user-images.githubusercontent.com/13201575/133863028-980a88cb-b267-4b7b-978d-0a35f51bf6a3.png)
